mne/viz/utils.py

```diff
@@ -67,6 +67,7 @@ def tight_layout(pad=1.2, h_pad=None, w_pad=None, fig=None):
 
     try:  # see https://github.com/matplotlib/matplotlib/issues/2654
         fig.canvas.draw()
+	fig.tight_layout(pad=pad, h_pad=h_pad, w_pad=w_pad)
         fig.set_tight_layout(dict(pad=pad, h_pad=h_pad, w_pad=w_pad))
```

Comment:
you need which call? I was talking about the call to `set_tight_layout`, I think you're talking about the call to `tight_layout` (which is the one that checks for compatibility). I think it makes the most sense to put only the `fig.tight_layout` call in the `try` part of this, since we're using it as the test, then put `set_tight_layout` in the `else` and `fig.canvas.draw()` before the exception catching at all.